build.sbt

```diff
@@ -851,3 +850,154 @@ commands ++= {
+        |        <option name="languageLevel" value="Scala_2_12" />
+        |        <compiler-classpath>
+        |$entries
+        |        </compiler-classpath>
+        |      </properties>
+        |      <CLASSES />
+        |      <JAVADOC />
+        |      <SOURCES />
+        |    </library>""".stripMargin
+  }
+
+  def replaceLibrary(data: Node, libName: String, libType: Option[String], newContent: String) = {
+    object rule extends RewriteRule {
+      var transformed = false
+      def checkAttrs(attrs: MetaData) = {
+        def check(key: String, expected: String) = {
+          val a = attrs(key)
+          a != null && a.text == expected
+        }
+        check("name", libName) && libType.forall(tp => check("type", tp))
+      }
+
+      override def transform(n: Node): Seq[Node] = n match {
+        case e @ Elem(_, "library", attrs, _, _, _*) if checkAttrs(attrs) =>
+          transformed = true
+          XML.loadString(newContent)
+        case other =>
+          other
+      }
+    }
+    object trans extends RuleTransformer(rule)
+    val r = trans(data)
+    if (!rule.transformed) error(s"Replacing library classpath for $libName failed, no existing library found.")
+    r
+  }
+
+  val ipr = (baseDirectory in ThisBuild).value / "src/intellij/scala.ipr"
+
+  var continue = false
+  if (!ipr.exists) {
+    scala.Console.print(s"Could not find src/intellij/scala.ipr. Create new project files from src/intellij/*.SAMPLE (y/n)? ")
+    if (scala.Console.readLine() == "y") {
+      intellijCreateFromSample((baseDirectory in ThisBuild).value)
+      continue = true
+    }
+  } else {
+    scala.Console.print("Update library classpaths in the current src/intellij/scala.ipr (y/n)? ")
+    continue = scala.Console.readLine() == "y"
+  }
+  if (continue) {
+    s.log.info("Updating library classpaths in src/intellij/scala.ipr.")
+    val content = XML.loadFile(ipr)
+
+    val newStarr = replaceLibrary(content, "starr", Some("Scala"), starrDep((scalaInstance in LocalProject("compiler")).value.jars))
+    val newModules = modules.foldLeft(newStarr)({
+      case (res, (modName, jars)) =>
+        if (jars.isEmpty) res // modules without dependencies
+        else replaceLibrary(res, s"$modName-deps", None, moduleDep(modName, jars))
+    })
+
+    XML.save(ipr.getCanonicalPath, newModules)
+  } else {
+    s.log.info("Aborting.")
+  }
+}
+
+lazy val intellijFromSample = taskKey[Unit]("Create fresh IntelliJ project files from src/intellij/*.SAMPLE.")
+
+intellijFromSample := {
+  val s = streams.value
+  scala.Console.print(s"Create new project files from src/intellij/*.SAMPLE (y/n)? ")
+  if (scala.Console.readLine() == "y")
+    intellijCreateFromSample((baseDirectory in ThisBuild).value)
+  else
+    s.log.info("Aborting.")
+}
+
+def intellijCreateFromSample(basedir: File): Unit = {
+  val files = basedir / "src/intellij" * "*.SAMPLE"
+  val copies = files.get.map(f => (f, new File(f.getCanonicalPath.stripSuffix(".SAMPLE"))))
```

Comment:
In places where you don't build a URI (and thus shouldn't deal with native path names anyway) I'd use `getAbsolutePath`, which may work in more cases and should be good enough for constructing a new `File`.